### PR TITLE
Design doc: Batch Normalization Operator

### PR DESCRIPTION
**Pull request by Canpio**
Related: #3684 
[Here](https://github.com/Canpio/Paddle/blob/e85c09fc7aa5209efbe2284ce99387de8913c6ad/paddle/operators/batch_norm_op.md) may be easier to read.

Some part of `batch_norm_op` design is strongly related with Python API and needs future discussions.

### CODE REVIEW

File: paddle/operators/batch_norm_op.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+```python 
+def batch_norm_layer(net, 
+                     input,
+                     output, 
+                     scale, 
+                     bias, 
+                     use_global_est = False, 
+                     epsilon = 1e-6,
+                     momentum = 0.99):
+	mean_cache = scope.new_var(name = 'estimated_mean', trainable = False)
```
Comment (Superjom) [inline paddle/operators/batch_norm_op.md]: There might be more than one batch_norm_op in the same topology, make sure variables like `mean_cache` and others have unique names?

Comment (Superjom) [reply paddle/operators/batch_norm_op.md]: make sure variables such as `mean_cache` invisible to other operators, or may be two operators write it.

File: paddle/operators/batch_norm_op.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+```python
+for pass_id in range(PASS_NUM):
+    # ...
+    net.train()  # run training model
+    if pass_id % 100 == 0:
+        net.infer(test_image)    # run inferencing model
+    # ...
+``` 
+
+`is_infer` is an attribute. Once an operator is created, its attributes can not be changed. It suggests us that we shall maintain two `batch_norm_op` in the model, one's `is_infer` is `True`(we call it `infer_batch_norm_op`) and the other one's is `False`(we call it `train_batch_norm_op`). They share all parameters and variables. How to organize them is related with Python API design, so I leave it here for further discussion.
```
Comment (Superjom) [inline paddle/operators/batch_norm_op.md]: Currently, different Block shares the same operator objects.  Did this require that different blocks has their own operator objects? 

Add some details here?

File: paddle/operators/batch_norm_op.md (hunk 1 above)
Comment (Canpio) [reply paddle/operators/batch_norm_op.md]: 1. Yes. we need to make sure `mean_cache` has the unique name.
2. `mean_cache` is defined inside `def batch_norm_layer`, could this make sure it's invisible for other operators?

File: paddle/operators/batch_norm_op.md (hunk 2 above)
Comment (Canpio) [reply paddle/operators/batch_norm_op.md]: We do need two distinct `batch_norm_op`. And it seems really require blocks holds their own operator objects... We shall have more discussion on it.

**Comment by Canpio** [comment]
This design is temporarily placed on hold for it's strongly related to Python API design.
